dask_expr/align.py

```diff
@@ -0,0 +1,51 @@
+import functools
+
+from tlz import merge_sorted, unique
+
+from dask_expr.expr import Expr, Projection, is_broadcastable
+from dask_expr.repartition import RepartitionDivisions
+
+
+class AlignDivisions(Expr):
```

Comment:
Thoughts on renaming to `AlignPartitions`?  

It feels like we're moving away from the term divisions in general.  Maybe partitions has more longevity.

Reply:
Renamed